### PR TITLE
avoid longjmps when restoring workspace

### PR DESCRIPTION
**Pull request by kevinushey**
Note: obviously risky but also fixes a pretty bad IDE hang on Windows with 64bit R when attempting to load a corrupt `.RData` file (due to `longjmp` madness).

This PR attempts to safely restore the user's workspace on startup. Note that:

1. `R_RestoreGlobalEnvironmentFromFile` is a C entrypoint that just calls the R code `sys.load.image()`,
2. That itself is just an R function that calls `load(file, envir = .GlobalEnv)` (and does some optional printing),
3. Calls to `load` can produce a (catchable) R error.

This PR avoids that R `longjmp` leaking out by catching the R error, and returning the error message as a string (or an empty string if everything succeeded).

cross-ref: https://github.com/wch/r-source/blob/trunk/src/main/saveload.c#L2174-L2195

### CODE REVIEW

File: src/cpp/r/session/RSession.cpp [hunk 1]
```diff
@@ -285,13 +292,22 @@ void deferredRestoreNewSession()
...
+      if (error)
       {
-         reportDeferredDeserializationError(error);
+         LOG_ERROR(error);
+      }
+      else if (!errMessage.empty())
+      {
+         std::stringstream ss;
+         std::string aliasedPath = createAliasedPath(globalEnvPath);
+         ss << "WARNING: Failed to restore workspace from '" << aliasedPath << "'.\n"
```
Comment (jmcphers) [inline src/cpp/r/session/RSession.cpp]: Nit: mix of `\n` and `std::endl`

File: src/cpp/r/session/RSession.cpp [hunk 2]
```diff
@@ -285,13 +292,22 @@ void deferredRestoreNewSession()
...
-                                        path.c_str(),
-                                        TRUE));
-      if (error)   
+      
+      std::string errMessage;
+      Error error = restoreGlobalEnvFromFile(path, &errMessage);
+      if (error)
       {
-         reportDeferredDeserializationError(error);
+         LOG_ERROR(error);
```
Comment (jmcphers) [inline src/cpp/r/session/RSession.cpp]: Should this error be shown to the user (as was done previously in `reportDeferredDeserializationError`)?

Comment (kevinushey) [inline src/cpp/r/session/RSession.cpp]: Good point; I'll make that change.

File: src/cpp/r/session/RSession.cpp (hunk 1 above)
Comment (kevinushey) [inline src/cpp/r/session/RSession.cpp]: Fixed that up; thanks!

**Comment by jmcphers** [comment]
Thanks, this looks good now.

**Comment by jjallaire** [comment]
Let's add a NEWS item for this if we haven't already

**Comment by kevinushey** [comment]
Added here: https://github.com/rstudio/rstudio/commit/5c2df33c514d6b5044ebbbf8babc83bacfbd6348

**Pull request merged**